app/build.gradle

```diff
@@ -94,6 +96,7 @@ android {
         alpha { java.srcDirs += 'src/extra/java' }
         beta { java.srcDirs += 'src/extra/java' }
         greatfire { java.srcDirs += 'src/extra/java' }
+        fdroid { java.srcDirs += 'src/extra/java' }
```

Comment:
I think it is possible to do something like this for the _res_ files, so that they don't need to be copied.  But copying seems fine too since its expedient.

Reply:
it looks like the src/extra/java directory is where the non-stubbed versions of the mlkit and firebase classes exist.  i assume by adding that path for every flavor but fdroid it includes the classes that require the proprietary dependencies in every non-fdroid flavor.